pkg/agent/params.go

```diff
@@ -220,15 +221,11 @@ func assignKubernetesParameters(properties *datamodel.Properties, parametersMap
 
 		if servicePrincipalProfile != nil {
 			addValue(parametersMap, "servicePrincipalClientId", servicePrincipalProfile.ClientID)
-			keyVaultSecretRef := servicePrincipalProfile.KeyvaultSecretRef
-			if keyVaultSecretRef != nil {
```

Comment:
Does removing this break the current scenarios?

Reply:
Confirmed with @norshtein , we have never support keyvalutref for SP in AKS. Also I  have a  self comment of this change